kratos/utilities/math_utils.h

```diff
@@ -827,6 +827,47 @@ class MathUtils
         return std::sqrt(temp);
     }
 
+    /**
+     * Calculates the norm of vector "a" while avoiding underflow and overflow.
+     * @param a Input vector
+     * @return The resulting norm
+     * @see http://www.netlib.org/lapack/explore-html/da/d7f/dnrm2_8f_source.html
+     */
+
+    static inline TDataType StableNorm(const Vector& a)
+    {
+        if (a.size() == 0) {
+            return 0;
+        }
+
+        if (a.size() == 1) {
+            return a[0];
+        }
+
+        TDataType max {0};
+
+        TDataType sqr_sum_scaled {1};
+
+        for (auto it = a.begin(); it != a.end(); ++it) {
+            TDataType x = *it;
+
+            if (x != 0) {
+                const TDataType abs_x = std::abs(x);
+
+                if (max < abs_x) {
+                    x = max / abs_x;
+                    sqr_sum_scaled = sqr_sum_scaled * (x * x) + 1.0;
```

Comment:
Where the +1..0 comes from?, it is not in the link for LAPACK you added (thanks :D )

Reply:
They wrote one avoid define multiple times 1.0d...

Reply:
The entries of the value are scaled according to the maximum. This line is executed when `x[i]` is the new maximum. So `x[i] / max = x[i] / x[i] = 1`. But I maybe should not reuse `x` in this scenario...